working/1426-extension-types/feature-specification.md

```diff
@@ -0,0 +1,948 @@
+}
+
+void main() {
+  MyInt m = 42;
+  m.twice; // OK, in the extension type.
+  m.isEven; // OK, a shown instance member.
+  m.ceil(); // OK, a shown instance member.
+  m.toString(); // OK, an `Object?` member.
+  m.floor(); // Error, hidden.
+}
+```
+
+
+### Implementing superinterfaces
+
+This section specifies the effect of having an `<interfaces>` part in an
+extension declaration.
+
+Let `E` be an extension declaration where `<interfaces>?` is of the form
+`implements T1, .. Tm`. We say that `E` has `T1` .. `Tm` as its direct
+superinterfaces.
+
+A compile-time error occurs if a direct superinterface does not denote a
+class, or if it denotes a class which cannot be a superinterface of a
+class.
+
+*For instance, `implements int` is an error.*
+
+If the `<interfaces>?` part of `E` is empty, it is treated as
+`implements Object`.
+
+For each member `m` named `n` in each direct superinterface of `E`, an
+error occurs unless `E` declares a member `m1` named `n` which is a correct
+override of `m`, or the show/hide part of `E` enables an instance member of
+the on-type which is a correct override of `m`.
+
+No subtype relationship exists between `E` and `T1, .. Tm`.
+
+*This means that when an extension type implements a set of interfaces, it
+is enforced that all the specified members are available, and that they
+have a signature which is compatible with the ones in `T1, .. Tm`, but
+there is no assignability from an expression of type `E` to a variable
+whose declared type is `Tj` for some `j` in 1..m. For that, it is necessary
+to use `box`, as described below.*
+
+
+### Gaining control over the instances: Protected extension types
+
+This section specifies the effect of including the keyword `protected` as
+the first token in an extension declaration.
+
+*The core idea is that no object can have a protected extension type as its
+static type unless it has been returned by an extension type constructor.
+This allows developers to gain control over which instances get to have
+that type.*
+
+Let `D` be an explicit extension type declaration named `E` which is
+prefixed by `protected`. We say that it is a _protected extension type
+declaration_, and that it introduces a _protected extension type_.
+
+Let `E` be the name of a protected extension type declaration.
+The type `E` (or
+<code>E<T<sub>1</sub>, .. T<sub>k</sub>></code>
+if `E` is generic) is a proper subtype of `Object?` and a proper supertype
+of `Never`.
+
+*In contrast to non-protected extension types, the extension type has no
+subtype relationship with its on-type.*
+
+It is a compile-time error if a protected extension type `E` is the
+target type in a type test (`o is E` respectively
+<code>o is E<T<sub>1</sub>, .. T<sub>k</sub>></code>)
+or a type cast (`o as E` respectively
+<code>o as E<T<sub>1</sub>, .. T<sub>k</sub>></code>).
+The type `dynamic` is not assignable to any protected extension type.
+
+*The subtype relationships and the type test/cast errors ensure that an
```

Comment:
This isn't true, unless there's something I'm missing here (e.g. you also don't allow these to be subtypes of `Object`.  See my example above.

Reply:
That example relied on a cast to a protected extension type, so unless we relax the rules and allow the abstraction to be violated in that manner then this should be true. In any case, I think we can handle this discussion based on the comment at line 635, so I'll resolve this one.